src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/AwsV2CloudUtil.java

```diff
@@ -0,0 +1,165 @@
+package cloud.fogbow.ras.core.plugins.interoperability.aws;
+
+import cloud.fogbow.common.exceptions.FogbowException;
+import cloud.fogbow.common.exceptions.InstanceNotFoundException;
+import cloud.fogbow.common.exceptions.UnexpectedException;
+import cloud.fogbow.ras.constants.Messages;
+import org.apache.log4j.Logger;
+import software.amazon.awssdk.core.exception.SdkException;
+import software.amazon.awssdk.services.ec2.Ec2Client;
+import software.amazon.awssdk.services.ec2.model.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AwsV2CloudUtil {
+
+    public static final Logger LOGGER = Logger.getLogger(AwsV2CloudUtil.class);
+    public static final Object SECURITY_GROUP_RESOURCE = "Security Groups";
+    public static final String AWS_TAG_GROUP_ID = "groupId";
+    public static final String TCP_PROTOCOL = "tcp";
+    public static final String AWS_TAG_NAME = "Name";
+    public static final Integer SSH_DEFAULT_PORT = 22;
+    
+    public static Image getImagesFrom(DescribeImagesResponse response) throws FogbowException {
+        if (response != null && !response.images().isEmpty()) {
+            return response.images().listIterator().next();
+        }
+        throw new InstanceNotFoundException(Messages.Exception.INSTANCE_NOT_FOUND);
+    }
+    
+    public static DescribeImagesResponse doDescribeImagesRequest(DescribeImagesRequest request, Ec2Client client)
+            throws FogbowException {
+        try {
+            return client.describeImages(request);
+        } catch (Exception e) {
+            throw new UnexpectedException(String.format(Messages.Exception.GENERIC_EXCEPTION, e), e);
+        }
+    }
+    
+    public static Volume getVolumeFrom(DescribeVolumesResponse response) throws FogbowException {
+        if (response != null && !response.volumes().isEmpty()) {
+            return response.volumes().listIterator().next();
+        }
+        throw new InstanceNotFoundException(Messages.Exception.INSTANCE_NOT_FOUND);
+    }
+    
+    public static DescribeVolumesResponse doDescribeVolumesRequest(Ec2Client client, DescribeVolumesRequest request)
+            throws FogbowException {
+        try {
+            return client.describeVolumes(request);
+        } catch (Exception e) {
+            throw new UnexpectedException(String.format(Messages.Exception.GENERIC_EXCEPTION, e), e);
+        }
+    }
+    
+    public static void createTagsRequest(String resourceId, String key, String value, Ec2Client client)
+            throws FogbowException {
+
+        Tag tag = Tag.builder()
+                .key(key)
+                .value(value)
+                .build();
+
+        CreateTagsRequest request = CreateTagsRequest.builder()
+                .resources(resourceId)
+                .tags(tag)
+                .build();
```

Comment:
don't we want to include the code between lines 59-67 in the try block?

Reply:
The build of Tag and CreateTagsRequest objects does not throw exceptions, just the calling a client's createTags method can throw some kind of SdkException. I'll change the exception of catching for SdkException that has as default in the other methods of this class.